osquery/core/windows/process.cpp

```diff
@@ -274,27 +277,35 @@ std::shared_ptr<PlatformProcess> PlatformProcess::launchExtension(
     return std::shared_ptr<PlatformProcess>();
   }
 
-  auto status = ::CreateProcessA(exec_path.c_str(),
-                                 mutable_argv.data(),
-                                 nullptr,
-                                 nullptr,
-                                 TRUE,
-                                 0,
-                                 nullptr,
-                                 nullptr,
-                                 &si,
-                                 &pi);
-  unsetEnvVar("OSQUERY_EXTENSION");
-
-  if (!status) {
-    return std::shared_ptr<PlatformProcess>();
-  }
+  auto ext_path = fs::path(exec_path);
+
+  // We are autoloading a Python extension, so pass off to our helper
+  if (ext_path.extension().string() == ".ext") {
```

Comment:
Maybe we can check for `.py` and allow `.ext` to be the general case like on other platforms?

Reply:
Well, for other plats if the extension doesn't end in `.ext` it doesn't get loaded. We don't check for `.py` on other plats, we assume the system knows how to `execv` the file we pass. This change was just attempting to make the Windows style reflect the posix setup.

Reply:
Maybe just remove this check?

Reply:
IIRC launchTestPythonScript is bad news, and should only be used for running our integration tests.

Reply:
I was going to propose the same as @theopolis . Every time a new language for extension is added, it will require core code changes. Might as well remove this check.

Reply:
@theopolis do you recall what the issue was? The code logic for launchTestPythonScript on Windows is nearly the same as launchExtension, with the exception that it attempts to derive the Python binary path before calling CreateProcess. Any thoughts on how we could get around this? I can roll another helper func to launch the extensions, but I’d bet the logic will more or less be similar.

Reply:
So, unfortunately this check can't exactly be removed, as we need a differentiation between `.exe` extensions and those which need a different method of exection. I can restructure the logic to just prepend the python binary path before we call `CreateProcess`, however this would still entail having this, or a similar check for the type of the file, and similar logic would be needed if we extend our extensions framework to support other languages which Windows can't invoke directly.

Reply:
This is a really high impact feature, is it worth landing this now (since it works as is) and opening an issue for discussion on what we should do to fix this longer term?